eva-server/src/main/java/uk/ac/ebi/eva/server/ws/EvaWSServer.java

```diff
@@ -78,52 +82,72 @@ public Jackson2ObjectMapperBuilder jacksonBuilder() {
     
     public EvaWSServer() { }
 
-    protected void initializeQueryOptions() {
+    protected void initializeQuery() {
+        startTime = System.currentTimeMillis();
+        initializeQueryOptions();
+    }
+
+    private void initializeQueryOptions() {
         this.queryOptions = new QueryOptions();
         Map<String, String[]> multivaluedMap = httpServletRequest.getParameterMap();
-        
+
         boolean metadata = (multivaluedMap.get("metadata") != null) ? multivaluedMap.get("metadata")[0].equals("true") : true ;
         int limit = (multivaluedMap.get("limit") != null) ? Integer.parseInt(multivaluedMap.get("limit")[0]) : -1;
         int skip = (multivaluedMap.get("skip") != null) ? Integer.parseInt(multivaluedMap.get("skip")[0]) : -1;
         boolean count = (multivaluedMap.get("count") != null) ? multivaluedMap.get("count")[0].equals("true") : false ;
 
         String[] exclude = multivaluedMap.get("exclude");
         String[] include = multivaluedMap.get("include");
-        
+
         queryOptions.put("metadata", metadata);
         queryOptions.put("exclude", (exclude != null && exclude.length > 0) ? Splitter.on(",").splitToList(exclude[0]) : null);
         queryOptions.put("include", (include != null && include.length > 0) ? Splitter.on(",").splitToList(include[0]) : null);
         queryOptions.put("limit", (limit > 0) ? limit : -1);
         queryOptions.put("skip", (skip > 0) ? skip : -1);
         queryOptions.put("count", count);
-        System.out.println(queryOptions.toJson());
+        logger.debug(queryOptions.toJson());
     }
 
-    protected QueryResponse setQueryResponse(Object obj) {
-        QueryResponse queryResponse = new QueryResponse();
-    	endTime = System.currentTimeMillis() - startTime;
-    	// TODO Restore span time calculation
-//        queryResponse.setTime(new Long(endTime - startTime).intValue());
+    protected <T> QueryResponse<T> setQueryResponse(T obj) {
+        QueryResponse<T> queryResponse = new QueryResponse<>();
+        endTime = System.currentTimeMillis();
         queryResponse.setApiVersion(version);
         queryResponse.setQueryOptions(queryOptions);
-        
-        // Guarantee that the QueryResponse object contains a coll of results
-        List coll;
-        if (obj instanceof List) {
-            coll = (List) obj;
-        } else {
-            coll = new ArrayList();
-            coll.add(obj);
-        }
+
+        // TODO why is this lost? because it's a native int?
```

Comment:
the dbtime in the queryResponse object is not received in the tests, that's why I added it to the queryResult